subsys/net/lib/websocket/websocket.c

```diff
@@ -531,7 +531,20 @@ static int sendmsg_all(int sock, const struct msghdr *message, int flags)
 
 	while (offset < total_len) {
 		ret = zsock_sendmsg(sock, message, flags);
-		if (ret < 0) {
+
+		if ((ret == 0) || (ret < 0 && errno == EAGAIN)) {
+			struct zsock_pollfd pfd;
+			int pollres;
+
+			pfd.fd = sock;
+			pfd.events = ZSOCK_POLLOUT;
+			pollres = zsock_poll(&pfd, 1, CONFIG_WS_CLIENT_SOCKET_POLL_TIMEOUT_MS);
```

Comment:
Same comment to websocket version, the `websocket_send_msg()` has also timeout option that we should honor.